solutions/cf-862B.mdx

```diff
@@ -0,0 +1,122 @@
+---
+id: cf-862B
+source: CF
+title: Bipartiteness
+author: Ryan Chou
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/54590) 
+
+## Explanation
+Given the number of colored and uncolored nodes, the maximum number of edges that can exist is $c \cdot uc$ where $c$ represents the number of colored nodes, and $uc$ represents the number of uncolored nodes. 
+
+This is because an edge can be drawn from each colored node to every other uncolored node. However, we already have $n - 1$ edges, so the answer is $(c \cdot uc) - (n - 1)$.
+
+To find the number of colored and uncolored nodes, we can run a [DFS](https://usaco.guide/silver/dfs), incrementing the number of colored and uncolored nodes as we traverse the graph.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N + M)$
+
+<LanguageSection>
+
+<JavaSection>
+
+```java
+import java.util.*;
+import java.io.*;
+
+public class Bipartiteness {
+	public static ArrayList<Integer> adj[];
+	
+	// counters for colored and uncolored nodes
+	public static long[] cnt = {0, 0};
```

Comment:
ryan
for those of us who have a dirty mind
please rename `cnt` to something else